lib/replicate.js

```diff
@@ -35,8 +35,27 @@ function replicate (feed, initiator, opts) {
         stream.close(feed.discoveryKey)
         return
       }
+      if (opts.onfeedauthenticate) {
+        if (!stream.remotePublicKey) {
+          stream.once('handshake', function () {
+            onready()
```

Comment:
it's not safe to call onready again as it that will mess with the ifAvailable counter

Reply:
If you call ifAvailable .wait() it might be fine I think

Reply:
Thank you for the feedback. I will try to address it, also 😨  I fear that won't be enough as the 'handshake' event may never occur (in case of error, or close) as such i probably need to listen/unlisten to other events as well?

Reply:
Yea it's tricky. You need to hook up `on(close)` as well and call the ifavailable hook but ONLY if the happy path hasn't succeded.